controllers/storagecluster/reconcile.go

```diff
@@ -163,6 +164,11 @@ func (r *StorageClusterReconciler) Reconcile(ctx context.Context, request reconc
 		return reconcile.Result{}, err
 	}
 
+	// Set default mon count value in the StorageCluster CR
+	if sc.Spec.ManagedResources.CephCluster.MonCount == 0 {
```

Comment:
You should also set the default on the API.

Reply:
Done